src/query/v2/requests.hpp

```diff
@@ -57,24 +57,25 @@ struct EdgeId {
   Gid gid;
 };
 
-struct Vertex {
-  VertexId id;
-  std::vector<Label> labels;
-  friend bool operator==(const Vertex &lhs, const Vertex &rhs) {
-    return (lhs.id == rhs.id) && (lhs.labels == rhs.labels);
-  }
-};
-
 struct Edge {
   VertexId src;
   VertexId dst;
+  std::optional<std::vector<std::pair<PropertyId, Value>>> properties;
   EdgeId id;
   EdgeType type;
   friend bool operator==(const Edge &lhs, const Edge &rhs) {
     return (lhs.src == rhs.src) && (lhs.dst == rhs.dst) && (lhs.type == rhs.type);
   }
 };
 
+struct Vertex {
+  VertexId id;
+  std::vector<Label> labels;
+  friend bool operator==(const Vertex &lhs, const Vertex &rhs) {
+    return (lhs.id == rhs.id) && (lhs.labels == rhs.labels);
```

Comment:
How is the order of labels ensured, since two vectors of the same labels can yield false here?

Reply:
I think this is a bug. `VertexId` should be `unique` since it's the primary label + primary key. I think for the comparison to work it sufficies to compare only the id's (`VertexId`)

Reply:
I did change the comparison.